XPlat/AzureMessaging/iOS/samples/NotificationHubsSample/AppDelegate.cs

```diff
@@ -14,104 +13,58 @@ namespace AzureMessagingSampleiOS
 
-			homeViewController = new HomeViewController (HUB_NAME == "YOUR-HUB-NAME");
+			var homeViewController = new HomeViewController(needsConfig);
 
 			// If you have defined a root view controller, set it here:
 			window.RootViewController = homeViewController;
 
-			// Process any potential notification data from launch
-			ProcessNotification (options);
+			MSNotificationHub.Init(CONNECTION_STRING, HUB_NAME);
 
-			// Register for Notifications
-			var notificationTypes = UIUserNotificationType.Badge
-									| UIUserNotificationType.Sound
-									| UIUserNotificationType.Alert;
+			Console.WriteLine("Device Token: " + MSNotificationHub.GetPushChannel());
 
-			var notificationSettings = UIUserNotificationSettings.GetSettingsForTypes (notificationTypes, null);
-
-			UIApplication.SharedApplication.RegisterUserNotificationSettings (notificationSettings);
+			MSNotificationHub.SetDelegate(new NotificationListener(homeViewController));
 
 			// make the window visible
 			window.MakeKeyAndVisible ();
 			
 			return true;
-		}
-
-		public override void DidRegisterUserNotificationSettings (UIApplication application, UIUserNotificationSettings notificationSettings)
-		{
-			UIApplication.SharedApplication.RegisterForRemoteNotifications ();
-		}
-
-		public override void ReceivedRemoteNotification (UIApplication application, NSDictionary userInfo)
-		{
-			// Process a notification received while the app was already open
-			ProcessNotification (userInfo);
-		}
+		}       
+	}
 
-		public override void RegisteredForRemoteNotifications (UIApplication application, NSData deviceToken)
-		{
-			// Connection string from your azure dashboard
-			var cs = SBConnectionString.CreateListenAccess(
-				new NSUrl("sb://" + HUB_NAME + "-ns.servicebus.windows.net/"),
-				HUB_LISTEN_SECRET);
+	public partial class NotificationListener : NSObject, IMSNotificationHubDelegate
+	{
+		HomeViewController homeViewController;
 
-			// Register our info with Azure
-			var hub = new SBNotificationHub (cs, HUB_NAME);
-			hub.RegisterNative (deviceToken, null, err => {
+		public NotificationListener(HomeViewController homeViewController)
+        {
+			this.homeViewController = homeViewController;
 
-				if (err != null) {
-					Console.WriteLine("Error: " + err.Description);
-					homeViewController.RegisteredForNotifications ("Error: " + err.Description);
-				} else  {
-					Console.WriteLine("Success");
-					homeViewController.RegisteredForNotifications ("Successfully registered for notifications");
-				}
-			});
 		}
 
-		void ProcessNotification(NSDictionary userInfo)
-		{
-			if (userInfo == null)
-				return;
-
-			Console.WriteLine ("Received Notification");
-
-			var apsKey = new NSString ("aps");
-				
-			if (userInfo.ContainsKey (apsKey)) {
-
-				var alertKey = new NSString ("alert");
+        public void DidReceivePushNotification(MSNotificationHub notificationHub, MSNotificationHubMessage message, CompletionHandler completionHandler)
```

Comment:
Why are we not calling the completion handler with the NoData as we do in other platforms?

Reply:
Fixed.